xskillscore/core/contingency.py

```diff
@@ -0,0 +1,787 @@
+from functools import wraps
+
+import numpy as np
+import xarray as xr
+
+from xhistogram.xarray import histogram as xhist
+
+__all__ = ['histogram', 'Contingency']
+
+OBSERVATIONS_NAME = 'observations'
+FORECASTS_NAME = 'forecasts'
+
+
+def histogram(*args, bins=None, bin_names=None, **kwargs):
+    """
+        Wrapper on xhistogram to deal with Datasets appropriately
+    """
+    if isinstance(args[0], xr.core.dataset.Dataset):
+        # Get list of variables that are shared across all Datasets
+        overlapping_vars = set.intersection(*map(set, [arg.data_vars for arg in args]))
+        if overlapping_vars:
+            # If bin_names not provided, use default ----
+            if bin_names is None:
+                bin_names = ['ds_' + str(i + 1) for i in range(len(args))]
+            return xr.merge(
+                [
+                    xhist(
+                        *(arg[var].rename(bin_names[i]) for i, arg in enumerate(args)),
+                        bins=bins,
+                        **kwargs,
+                    ).rename(var)
+                    for var in overlapping_vars
+                ]
+            )
+        else:
+            raise ValueError('No common variables exist across input Datasets')
+    else:
+        if bin_names:
+            args = (arg.rename(bin_names[i]) for i, arg in enumerate(args))
+        return xhist(*args, bins=bins, **kwargs)
+
+
+def get_category_bounds(category_edges):
+    """
+        Return formatted string of cateogry bounds given list of category edges
+    """
+    return [
+        '(' + str(category_edges[i]) + ', ' + str(category_edges[i + 1]) + ']'
```

Comment:
I would suggest using f-strings here. Way more readable, and an awesome python3 thing.

```python
 '(' + str(category_edges[i]) + ', ' + str(category_edges[i + 1]) + ']'
```

to

```python
f"[{str(category_edges[i])}, {str(category_edges[i + 1])}]
```